CondFormats/SiPixelTransient/src/SiPixelTemplate2D.cc

```diff
@@ -1149,7 +1149,7 @@ bool SiPixelTemplate2D::xytemp(float xhit, float yhit, bool ydouble[BYM2], bool
       //combine rows and columns to simulate double pixels
       
       for(int m=1; m<BXM3; ++m) {
-         if(xdouble[m-1]) {
+         if(xdouble[m]) {
             //  Combine the x-rows
             for(int n=1; n<BYM3; ++n) {
                dxytempdx[1][m][n] += dxytempdx[1][m+1][n];

```

Comment:
here and below check of indexes is different is "m+1" always inside array size?

Reply:
Not sure this is the optimal way to deal with it, however ```m``` in ```dxytempdx``` is bound to ```BYM2```
https://github.com/OzAmram/cmssw/blob/94e74992470d17f6fdf3c12caef9712ce034f439/CondFormats/SiPixelTransient/src/SiPixelTemplate2D.cc#L1079
which corresponds to ```TYSIZE+2```
https://github.com/OzAmram/cmssw/blob/94e74992470d17f6fdf3c12caef9712ce034f439/CondFormats/SiPixelTransient/interface/SiPixelTemplateDefs.h#L28

```BXM3``` is ```TYSIZE+1```
https://github.com/OzAmram/cmssw/blob/94e74992470d17f6fdf3c12caef9712ce034f439/CondFormats/SiPixelTransient/interface/SiPixelTemplateDefs.h#L29
and therefore it does not go beyond the edge of the array by construction

Reply:
Ok, then extra question: why in this class array are accessed by value in several methods and not by reference?

Reply:
what do you mean with ``array by value"?  These are C-style arrays, if I am not mistaken....

Reply:
@civanch 
I don't know.  The code is from Morris Swartz.  We will need to ask him.  I think the best would be to integrate this PR (since the code runs), so that we can integrate other changes (e.g. in FastSim) that depends on the header change.  Then we can discuss this with Morris and make another PR just to address the sub-optimal performance.  If you'll be at the CMS week, the best would be to sit down with Morris and discuss all these changes in person.